### DISCUSSION TITLE
how to check classic queue version 2 setting

### DISCUSSION THREAD

**Discussion by 0Ams** [rabbitmq/rabbitmq-server#10137] [Questions] [answered]
I set classic version 2 by rabbitmq.conf

but, managment-ui not changed classic2 on queue's view

<img width="948" alt="image" src="https://github.com/rabbitmq/rabbitmq-server/assets/10244446/2d835df5-2f1d-48e5-adf0-e27a1da2a656">

how to check setting for classic queue default version 2?

```
$ rabbitmq-diagnostics environment | grep classic_queue_default_version
      {classic_queue_default_version,2},
```
It was confirmed that the value of env confirmed through A was changed.
Is there any other way to check that the default classic version has changed other than the environmental variables?

**Comment by michaelklishin** [comment, marked as answer]
`rabbitmq-diagnostics environment` is exactly how you inspect [effective node configuration](https://rabbitmq.com/configure.html#verify-configuration-effective-configuration).

Updating default queue version in node config will not update it for existing queues. The setting is only used at queue declaration time but you seem to believe that it will apply immediately.

That's not the case: none of the settings in configuration files [are applied on the fly](https://rabbitmq.com/configure.html#config-changes-effects). That's what [policies and runtime parameters](https://rabbitmq.com/parameters.html) are for.

CQv2 can be [adopted via a policy key](https://rabbitmq.com/persistence-conf.html#queue-version).

**Reply by 0Ams** [reply]
I understood the setting through [adopted via a policy key](https://rabbitmq.com/persistence-conf.html#queue-version). 
However, I wonder if the basic classic queue works as version 2 when set as `{classic_queue_default_version,2}` on rabbitmq.conf.

Can I get the same effect as the value I set through CQv2 adopted via a policy key? 
Do the two settings mean the same thing?

**Reply by mkuratczyk** [reply]
These are two methods of making a queue v2. The policy is more flexible - you can pick which queues you want as v2. On the other hand if you don't change the default, you need to include the `x-queue-version=2` in every policy you have where you want v2 (since only one policy is active).

3.13 will make v2 the default anyway.

**Reply by michaelklishin** [reply]
I don't know what a "basic classic queue" is, sorry. The queue storage version used is displayed on the individual queue page in the management UI.

**Reply by 0Ams** [reply]
a "basic classic queue" in my question meant classic in the screenshot.
![image](https://github.com/rabbitmq/rabbitmq-server/assets/10244446/cf4174be-cfd6-47c7-b78f-304dabec3f33)

What I want is to set up for all queues without policy specification.
When I set it up like "{classic_queue_default_version,2}", I want to check if it's being used well as default CQ2, but I don't know where or how to check it.

Should I just understand that CQ2 is being used well in rabbitmq@3.12.x, if the settings are well set up?

**Reply by mkuratczyk** [reply]
With recent versions, if you select a specific queue, you will see whether it's v1 or v2.

**Reply by michaelklishin** [reply]
> The queue storage version used is displayed on the individual queue page in the management UI
